src/hotspot/share/gc/shenandoah/shenandoahLock.cpp

```diff
@@ -32,40 +32,51 @@
 #include "runtime/javaThread.hpp"
 #include "runtime/os.inline.hpp"
 
-// These are inline variants of Thread::SpinAcquire with optional blocking in VM.
-
-class ShenandoahNoBlockOp : public StackObj {
-public:
-  ShenandoahNoBlockOp(JavaThread* java_thread) {
-    assert(java_thread == nullptr, "Should not pass anything");
-  }
-};
-
 void ShenandoahLock::contended_lock(bool allow_block_for_safepoint) {
   Thread* thread = Thread::current();
-  if (allow_block_for_safepoint && thread->is_Java_thread()) {
-    contended_lock_internal<ThreadBlockInVM>(JavaThread::cast(thread));
+  if (thread->is_Java_thread() && allow_block_for_safepoint) {
```

Comment:
Please flip `&&` back. It is easier to short-cut test after a boolean test we know is likely to constant-fold.

Reply:
Nice catch, thanks!